apollo/lib/websocket_manager.py

```diff
@@ -0,0 +1,90 @@
+import uuid
+
+from fastapi import WebSocket
+from starlette.websockets import WebSocketDisconnect
+from typing import Dict
+
+from apollo.lib.singleton import Singleton
+
+
+class WebSocketManager(metaclass=Singleton):
+    def __init__(self):
+        self.connections: Dict[uuid.UUID, WebSocket] = {}
+
+    @staticmethod
+    async def _keep_connection_open(websocket: WebSocket):
+        try:
+            while True:
+                await websocket.receive_json()
+        except WebSocketDisconnect:
+            return
+
+    @staticmethod
+    async def _send_message(target_websocket, message):
+        await target_websocket.send_json(message)
+
+    @staticmethod
+    def _check_runtime_error(error, message):
```

Comment:
This method name describes its use case but not what the method does from its own perspective. From its own perspective this method just checks whether `message` is in the stringified version of the error argument, which should be called `exception`.  If not it raises said exception.

It knows nothing about runtime or whatever. That’s just what caused you to create the method but it’ll work with any exception.